concordium-consensus/src/Concordium/Scheduler/Environment.hs

```diff
@@ -376,6 +376,9 @@ class (StaticInformation m, ContractStateOperations m, MonadProtocolVersion m) =
 
   -- |Transfer a scheduled amount from the first address to the second and run
   -- the computation in the modified environment.
+  --
+  -- Precondition: The list of releases MUST be non-empty and the timestamps MUST be in increasing
```

Comment:
Regarding this and `addScheduledAmountToCS`. 
I do not have a complete overview, but have you considered enforcing the non empty list  on the type level?

In that way we wouldn't be relying on the documentation (even it's fine to do so, that's why they are there in the first place :) ) 

But we could fail earlier than in `fromAccountReleaseScheduleV0#mkEntry` then.

Reply:
I think that's a fairly good idea, but I won't do it in this PR. In fact, for these functions it's OK for the list to be empty (it will just do nothing), and I have updated the documentation accordingly.

Reply:
Great!